contracts/Uniswap/UniswapLib.sol

```diff
@@ -1,68 +1,291 @@
 // SPDX-License-Identifier: GPL-3.0
 
-pragma solidity ^0.6.12;
+pragma solidity ^0.8.7;
 
-// Based on code from https://github.com/Uniswap/uniswap-v2-periphery
+// From: https://github.com/Uniswap/uniswap-v3-core
```

Comment:
Can we use their [recommended instructions with NPM](https://github.com/Uniswap/v3-core#local-deployment) instead of just referencing the repo? Then we can lock to a version.

Reply:
I believe auditors do like when dependencies are all extracted out into the repo like this. This was the main reason it was done this way (please correct me if I'm wrong @alexroan). In the end, I also had to make changes to Uniswap's math library to upgrade to solc 0.8 (stricter rules about not being able to negate uint256s), so would have had to do it anyway, at least for the `FullMath` lib.